### PR TITLE
Gtp improvements

### PR DESCRIPTION
**Pull request by iopq**
Making sure we don't crash, adding more tests, adding error messaging

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,9 +15,5 @@ rand            = "*"
...
 regex_macros    = "*"
 rustc-serialize = "*"
-# Temporarily use the github time crate, due to a Windowns bug in the released version
-# time            = "*"
-
-[dependencies.time]
-
-git = "https://github.com/rust-lang/time"
\ No newline at end of file
+time            = "*"
```
Comment (ujh) [inline Cargo.toml]: This is not really important at all, but can we please keep the dependencies in alphabetical order. Slight OCD speaking here. ;)

Comment (iopq) [inline Cargo.toml]: done

File: src/gtp/driver.rs [hunk 2]
```diff
@@ -72,6 +62,7 @@ impl Driver {
                 Command::TimeLeft           => print!("= \n\n"),
                 Command::TimeSettings       => print!("= \n\n"),
                 Command::Version            => print!("= {}\n\n", engine_version),
+                Command::ErrorMessage(e)    => print!("? {}\n\n", e),
```
Comment (ujh) [inline src/gtp/driver.rs]: Would it be possible to get rid of the `_` match case? Then the compiler could check here too if the match is exhaustive.

Comment (iopq) [inline src/gtp/driver.rs]: Done

**Pull request merged**